test/performance/framework/set_up.go

```diff
@@ -0,0 +1,237 @@
+	}
+	if err := wait.PollUntilContextCancel(ctx, config.WaitInterval, true, func(ctx context.Context) (bool, error) {
+		podList, err := kClient.CoreV1().Pods(ns).List(ctx, metav1.ListOptions{LabelSelector: client_pod.ScaleClientPodTemplateName})
+		if err != nil {
+			return false, nil
+		}
+		for _, pod := range podList.Items {
+			if pod.Status.PodIP == "" {
+				return false, nil
+			}
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("error when waiting scale test clients to get IP: %w", err)
+	}
+	return nil
+}
+
+func checkNodeStatus(node corev1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func validScaleSpecification(c *config.ScaleConfiguration) error {
+	if c.NpNumPerNs*2 > c.PodsNumPerNs {
+		return fmt.Errorf("networkPolicy quantity is too larger than 1/2 workload Pods quantity, scale may fail")
+	}
+	if c.SvcNumPerNs*2 > c.PodsNumPerNs {
+		return fmt.Errorf("service quantity is too larger than 1/2 workload Pods quantity, scale may fail")
+	}
+	return nil
+}
+
+func ScaleUp(ctx context.Context, kubeConfigPath, scaleConfigPath, templateFilesPath string) (*ScaleData, error) {
+	var td ScaleData
+	scaleConfig, err := config.ParseConfigs(scaleConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	klog.InfoS("Scale config", "scaleConfig", scaleConfig)
+	td.Specification = scaleConfig
+
+	td.templateFilesPath = templateFilesPath
+
+	if err := validScaleSpecification(&scaleConfig.ScaleConfiguration); err != nil {
+		return nil, err
+	}
+
+	kubeConfig, err := runtime.ResolveKubeconfig(kubeConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("error when retrieving incluster kubeconfig: %w", err)
+	}
+	td.kubeconfig = kubeConfig
+	kClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error when creating kubernetes client: %w", err)
+	}
+
+	controlPlaneNodes, err := kClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/control-plane="})
+	if err != nil {
+		return nil, fmt.Errorf("error when getting Nodes in the cluster: %w", err)
+	}
+	masterNodes, err := kClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master="})
+	if err != nil {
+		return nil, fmt.Errorf("error when getting Nodes in the cluster: %w", err)
+	}
+	if len(masterNodes.Items) == 0 && len(controlPlaneNodes.Items) == 0 {
+		return nil, fmt.Errorf("can not find a master/control-plane Node in the cluster")
+	}
+	for _, node := range masterNodes.Items {
+		td.controlPlaneNodes = append(td.controlPlaneNodes, node.Name)
+	}
+	for _, node := range controlPlaneNodes.Items {
+		td.controlPlaneNodes = append(td.controlPlaneNodes, node.Name)
+	}
+	klog.InfoS("List all ControlPlane Nodes in cluster", "controlPlaneNodes", td.controlPlaneNodes)
```

Comment:
I remembered that when querying based on multiple labels, by default, the matching results of these labels are taken as an intersection (AND) rather than a union (OR).